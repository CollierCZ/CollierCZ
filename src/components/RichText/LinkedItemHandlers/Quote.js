import { css } from "styled-components";
import { Chat } from "@kiwicom/orbit-components/lib/icons";
import { Text } from "@kiwicom/orbit-components";
import PropTypes from "prop-types";
import React from "react";
import parseHTML, { domToReact } from "html-react-parser";

const Quote = ({ quote }) => {
  const source = quote.elements.source.value
    ? `<footer class="source">${quote.elements.source.value}</footer>`
    : "";
  const parseSource = {
    replace: ({ attribs, children, name, type }) => {
      if (type === "tag") {
        if (name === "em") {
          return <cite>{domToReact(children, parseSource)}</cite>;
        }
      }
      if (!attribs) return;
      if (attribs.class === "source") {
        return (
          <footer>
            <Text element="div" type="secondary">
              {domToReact(children, parseSource)}
            </Text>
          </footer>
        );
      }
    }
  };
  const parseQuote = {
    replace: ({ children, name, type }) => {
      if (type === "tag") {
        if (name === "p") {
          return <Text>{domToReact(children, parseQuote)}</Text>;
        }
      }
    }
  };
  const quoteText = parseHTML(
    quote.elements.quote.value,
    parseQuote
  );
  const sourceText = parseHTML(source, parseSource);
  return (
    <div
      css={css`
        display: flex;
        align-items: center;
      `}
    >
      <Chat size="large" />
      <blockquote>
        {quoteText}
        {sourceText}
      </blockquote>
    </div>
  );
};

export default Quote;

Quote.propTypes = {
  quote: PropTypes.object.isRequired
};
