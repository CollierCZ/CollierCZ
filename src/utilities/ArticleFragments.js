import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`