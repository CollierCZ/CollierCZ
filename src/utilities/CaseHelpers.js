const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}