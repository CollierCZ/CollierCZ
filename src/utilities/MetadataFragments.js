import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on KontentItemHome {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on KontentItemHome {
    elements {
      socialmedia {
        value
      }
    }
  }
`