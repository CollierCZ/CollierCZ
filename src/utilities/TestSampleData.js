export const author = {
  short_bio: {
    value:
      '<p>Originally from <a href="https://en.wikipedia.org/wiki/Lexington,_Massachusetts" data-new-window="true" target="_blank" rel="noopener noreferrer">Lexington</a> and now residing in <a href="https://www.gotobrno.cz/en/" data-new-window="true" target="_blank" rel="noopener noreferrer">Brno</a>, Aaron is an educator, editor, and elucidator.</p>'
  },
  picture: {
    value: [
      {
        url:
          "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/f7377b84-8ff0-4b77-bc0f-cfdc34d379cc/quixote.jpg"
      }
    ]
  },
  name: {
    value: "Aaron Collier"
  },
  url: {
    value: "aaron-collier"
  }
};

export const simpleImage = {
  description: "A cool picture",
  url: "https://example.com/"
};

export const sampleQuote = {
  system: {
    type: "quote"
  },
  elements: {
    quote: {
      resolvedData: {
        html:
          "<p>You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.</p>"
      }
    },
    source: {
      resolvedData: {
        html:
          "<p>Atticus Finch in <em>To Kill a Mockingbird </em>by Harper Lee</p>"
      }
    }
  }
};

export const sampleShowcase = {
  system: {
    type: "showcase"
  },
  elements: {
    items: {
      linked_items: [
        {
          id: "457d3938-5907-5cca-947d-bae1583e105d",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=vj7TyUgBEW8"
            },
            name: {
              value: "Write the Docs talk"
            },
            short_description: {
              links: [
                {
                  linkId: "3cc9d486-53e2-495d-9e2b-25665e0ff29f",
                  urlSlug: "seeing-your-docs-through-different-eyes",
                  type: "article",
                  codename: "write_the_docs_2019"
                }
              ],
              resolvedData: {
                html:
                  '<p>A video of a talk I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference. Or read a <a data-item-id="3cc9d486-53e2-495d-9e2b-25665e0ff29f" href="">text version of the talk</a>.</p>'
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/51e9be55-8502-40ca-b374-a1305b36dc56/Slide16.PNG"
                }
              ]
            }
          },
          fields: {
            slug: "write-the-docs-talk"
          }
        },
        {
          id: "fce4fe5b-59c9-5c88-8634-16b63df49d4a",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=MlMkAqxoPZM"
            },
            name: {
              value: "Structured content video"
            },
            short_description: {
              links: [],
              resolvedData: {
                html:
                  "<p>A video about what structured content is and how it can benefit content creators.</p>"
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/b0481084-18a8-4cf0-ae01-d1a0dbd766ce/structured_content.jpg"
                }
              ]
            }
          },
          fields: {
            slug: "structured-content-video"
          }
        }
      ]
    }
  }
};
