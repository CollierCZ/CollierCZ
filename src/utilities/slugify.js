let slugify = (text) => {
    return text
        .toString()
        .toLowerCase()
        .trim()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/[^\w-]+/g, '')       // Remove all non-word chars
}


module.exports.slugify = slugify