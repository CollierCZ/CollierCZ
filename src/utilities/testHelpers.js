import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
