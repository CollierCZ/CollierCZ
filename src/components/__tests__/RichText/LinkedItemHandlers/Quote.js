import React from "react";
import Quote from "../../../RichText/LinkedItemHandlers/Quote";
import { sampleQuote } from "../../../../utilities/TestSampleData"
import { simpleRenderer } from "../../../../utilities/testHelpers";

describe("Quote", () => {
  it("renders a quote correctly", () => {
    simpleRenderer(<Quote quote={sampleQuote} />);
  });
});