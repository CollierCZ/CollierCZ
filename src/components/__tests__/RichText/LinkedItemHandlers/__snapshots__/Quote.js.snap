// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders a quote correctly 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
}

.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #5F738C;
  line-height: 20px;
  text-align: left;
  margin: 0;
}

.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<div
  css={
    Array [
      "
        display: flex;
        align-items: center;
      ",
    ]
  }
>
  <svg
    className="c0"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M20.696 4.8H3.304C2.584 4.8 2 5.384 2 6.104V17.89a1.269 1.269 0 001.304 1.259h2.609v3.478a.435.435 0 00.683.357l5.54-3.835h8.56c.72 0 1.304-.584 1.304-1.304V6.103c0-.72-.584-1.304-1.304-1.304zm-2.609 7.341a1.304 1.304 0 11-2.609 0 1.304 1.304 0 012.609 0zm-12.174 0a1.304 1.304 0 112.609 0 1.304 1.304 0 01-2.609 0zm4.783 0a1.304 1.304 0 112.608 0 1.304 1.304 0 01-2.608 0z"
    />
  </svg>
  <blockquote>
    <p
      className="c1"
    >
      You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.
    </p>
    <footer>
      <p
        className="c2"
      >
        <p>
          Atticus Finch in 
          <cite>
            To Kill a Mockingbird 
          </cite>
          by Harper Lee
        </p>
      </p>
    </footer>
  </blockquote>
</div>
`;
